File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/library/common/order/SelectOrderView.java
package com.github.anrimian.musicplayer.ui.library.common.order;

import com.github.anrimian.musicplayer.domain.models.order.Order;
import com.github.anrimian.musicplayer.domain.models.order.OrderType;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.Skip;

public interface SelectOrderView extends MvpView {

    @Skip
    void close(Order order);

    @AddToEndSingle
    void showSelectedOrder(OrderType order);

    @AddToEndSingle
    void showReverse(boolean selected);

    @AddToEndSingle
    void showFileNameEnabled(boolean checked);
}

File: app/src/main/java/com/github/anrimian/musicplayer/ui/library/common/order/SelectOrderView.kt
package com.github.anrimian.musicplayer.ui.library.common.order

import com.github.anrimian.musicplayer.domain.models.order.Order
import com.github.anrimian.musicplayer.domain.models.order.OrderType
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle
import moxy.viewstate.strategy.alias.Skip

interface SelectOrderView : MvpView {

    @Skip
    fun close(order: Order)

    @AddToEndSingle
    fun showSelectedOrder(orderType: OrderType)

    @AddToEndSingle
    fun showReverse(selected: Boolean)

    @AddToEndSingle
    fun showFileNameEnabled(checked: Boolean)

}